### PR TITLE
feat: support ZRS disk

### LINKED ISSUE #573: ZRS disk type support
**Is your feature request related to a problem?/Why is this needed**

**Describe the solution you'd like in detail**

With ZRS disk type support, we could solve azure disk limitations on AKS:
https://docs.microsoft.com/en-us/azure/aks/availability-zones#azure-disks-limitations

ZRS disk is supported from azure-sdk-for-go v53.0.0:
https://github.com/Azure/azure-sdk-for-go/blob/master/services/compute/mgmt/2020-12-01/compute/enums.go#L273-L282

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by andyzhangx**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
feat: support ZRS disk

To support ZRS disk on one subscription, run:
```console
az feature register --name SsdZrsManagedDisks --namespace Microsoft.Compute
az feature list -o table --query "[?contains(name, 'Microsoft.Compute/SsdZrsManagedDisks')].{Name:name,State:properties.state}"
az provider register --namespace Microsoft.Compute
```

 - `nodeAffinity` of ZRS disk
```
  spec:
    accessModes:
    - ReadWriteOnce
    capacity:
      storage: 10Gi
    claimRef:
      apiVersion: v1
      kind: PersistentVolumeClaim
      name: persistent-storage-statefulset-azuredisk-0
      namespace: default
      resourceVersion: "48991"
      uid: 47534b37-c817-464a-8b8e-c37427f77c00
    csi:
      driver: disk.csi.azure.com
      volumeAttributes:
        skuname: StandardSSD_ZRS
        storage.kubernetes.io/csiProvisionerIdentity: 1618402610023-8081-disk.csi.azure.com
      volumeHandle: /subscriptions/xxx/resourceGroups/mc_andy-aks1199_andy-aks1199_eastus2euap/providers/Microsoft.Compute/disks/pvc-47534b37-c817-464a-8b8e-c37427f77c00
    nodeAffinity:
      required:
        nodeSelectorTerms:
        - matchExpressions:
          - key: topology.disk.csi.azure.com/zone
            operator: In
            values:
            - eastus2euap-1
        - matchExpressions:
          - key: topology.disk.csi.azure.com/zone
            operator: In
            values:
            - eastus2euap-2
        - matchExpressions:
          - key: topology.disk.csi.azure.com/zone
            operator: In
            values:
            - eastus2euap-3
    persistentVolumeReclaimPolicy: Delete
    storageClassName: managed-csi
    volumeMode: Filesystem
  status:
    phase: Bound
```

**Which issue(s) this PR fixes**:

Fixes #573

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
feat: support ZRS disk
```

### CODE REVIEW

**Comment by andyzhangx** [comment]
BTW, cannot add ZRS e2e test since it's only supported in canary region, will add e2e test when it supports westus2 region later.

**Comment by andyzhangx** [comment]
retest

**Comment by edreed** [comment]
It would still be worthwhile to have e2e tests under a dedicated context, say `[zrs]`, event though we can;t currently stand up a pipeline. At least we would be able to run them manually against a cluster set up in a subscription that is enabled for ZRS.

File: pkg/azuredisk/controllerserver.go [hunk 1]
```diff
@@ -216,7 +216,23 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
...
+			Segments: map[string]string{topologyKey: diskZone},
+		},
+	}
+	if skuName == compute.StandardSSDZRS || skuName == compute.PremiumZRS {
+		klog.V(2).Infof("diskZone(%s) is reset as empty since disk(%s) is ZRS(%s)", diskZone, diskName, skuName)
+		diskZone = ""
+		accessibleTopology = []*csi.Topology{}
+		for i := 1; i <= 3; i++ {
+			topology := &csi.Topology{
+				Segments: map[string]string{topologyKey: fmt.Sprintf("%s-%d", d.cloud.Location, i)},
```
Comment (feiskyer) [inline pkg/azuredisk/controllerserver.go:231]: skip the `diskZone` which has already been added in L222?

Comment (andyzhangx) [reply pkg/azuredisk/controllerserver.go:231]: the logic is a little confusing which is still correct, let me make it more clear.

Comment (andyzhangx) [reply pkg/azuredisk/controllerserver.go:227]: fixed

**Comment by andyzhangx** [comment]
> It would still be worthwhile to have e2e tests under a dedicated context, say `[zrs]`, event though we can;t currently stand up a pipeline. At least we would be able to run them manually against a cluster set up in a subscription that is enabled for ZRS.

file an issue to track: https://github.com/kubernetes-sigs/azuredisk-csi-driver/issues/795, Once ZRS disk is ready on westus2 region(end of April), we could add ZRS disk e2e test

**Pull request merged**